File: VB/Scaffolding.ValidationErrors/Common/ViewModel/ISingleObjectViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Linq.Expressions
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.DataAnnotations
Imports Scaffolding.ValidationErrors.Common.Utils
Imports Scaffolding.ValidationErrors.Common.DataModel
Imports MessageBoxButton = System.Windows.MessageBoxButton
Imports MessageBoxImage = System.Windows.MessageBoxImage
Imports MessageBoxResult = System.Windows.MessageBoxResult

Namespace Scaffolding.ValidationErrors.Common.ViewModel
	Public Interface ISingleObjectViewModel(Of TEntity, TPrimaryKey)
		ReadOnly Property Entity() As TEntity
		ReadOnly Property PrimaryKey() As TPrimaryKey
	End Interface
	Public Interface IDetailEntityInfo
	End Interface
	Public Class DetailEntityInfo(Of TDetailEntity As Class)
		Implements IDetailEntityInfo
		Private privateDetailEntityKey As Object
		Public Property DetailEntityKey() As Object
			Get
				Return privateDetailEntityKey
			End Get
			Private Set(ByVal value As Object)
				privateDetailEntityKey = value
			End Set
		End Property
		Public Sub New(ByVal detailEntityKey As Object)
			Me.DetailEntityKey = detailEntityKey
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.ValidationErrors/Common/ViewModel/ISingleObjectViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Linq.Expressions;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.DataAnnotations;
using Scaffolding.ValidationErrors.Common.Utils;
using Scaffolding.ValidationErrors.Common.DataModel;
using MessageBoxButton = System.Windows.MessageBoxButton;
using MessageBoxImage = System.Windows.MessageBoxImage;
using MessageBoxResult = System.Windows.MessageBoxResult;

namespace Scaffolding.ValidationErrors.Common.ViewModel {
    public interface ISingleObjectViewModel<TEntity, TPrimaryKey> {
        TEntity Entity { get; }
        TPrimaryKey PrimaryKey { get; }
    }
    public interface IDetailEntityInfo { }
    public class DetailEntityInfo<TDetailEntity> : IDetailEntityInfo
        where TDetailEntity : class {
        public object DetailEntityKey { get; private set; }
        public DetailEntityInfo(object detailEntityKey) {
            this.DetailEntityKey = detailEntityKey;
        }
    }
}